### PR TITLE
feat: invite witnesses

### PR DESCRIPTION
**Pull request by bddjong**
This change adds support to invite witnesses for your wedding. In summary
- Added continue without inviting button
- There is no minimum amount of witnesses on the page
- The invite witnesses button is disabled if no witnesses are entered
- If a witness name is entered but not their email, or vice versa, the fieldset is invalidated (see photo)

image

image

### CODE REVIEW

File: pages/voorgenomen-huwelijk/getuigen/index.tsx [hunk 1]
```diff
@@ -155,3 +173,68 @@ export default function MultistepForm1() {
...
+        </p>
+        {formState.errors.witnesses?.[index]?.name?.type === "required" && (
+          <FormFieldDescription invalid>{t("form:name-required")}</FormFieldDescription>
+        )}
+        <Textbox
+          id={nameId}
+          type="text"
+          autoComplete={`name ${witnessId}`}
+          invalid={nameInvalid}
+          {...register(`witnesses.${index}.name`)}
```
Comment (Yolijn) [inline pages/voorgenomen-huwelijk/getuigen/index.tsx:220]: if you add the `required` option with the  error message you'll want to show you don't have to create your own errors later. https://react-hook-form.com/docs/useform/register 

So 
```js
{...register(`witnesses.${index}.name`, {required: 'some error message'})}
```

File: pages/voorgenomen-huwelijk/getuigen/index.tsx [hunk 2]
```diff
@@ -27,74 +29,75 @@ import {
...
-  };
+  const onWitnessSubmit = (formData: WitnessFormData) => {
+    if (!formData.witnesses) {
+      onContinueWithoutInvite();
+      return;
+    }
+
+    let hasError = false;
+
+    formData.witnesses.forEach((witness, index) => {
```
Comment (Yolijn) [inline pages/voorgenomen-huwelijk/getuigen/index.tsx:78]: I think these errors can already be added during the form-input registration. See comment below.

File: pages/voorgenomen-huwelijk/getuigen/index.tsx (hunk 1 above)
Comment (Yolijn) [reply pages/voorgenomen-huwelijk/getuigen/index.tsx:220]: or you can just use `required: true` and keep the error message in the invalid form-field description

**Review by Yolijn — CHANGES_REQUESTED** [review]

File: pages/voorgenomen-huwelijk/getuigen/index.tsx (hunk 1 above)
Comment (bddjong) [reply pages/voorgenomen-huwelijk/getuigen/index.tsx:220]: It's a bit iffy here though... we don't necessarily want witnesses to be required, but rather, we want one of the fields to be required only if the related `email` or `name` field have any values. Setting required as above would result in the browser forcing a value regardless, even though the user might not want to add all witnesses.

Comment (bddjong) [reply pages/voorgenomen-huwelijk/getuigen/index.tsx:220]: I thought of a few other ways to solve this. One would be to check during the render if the `name` field has any value, set `email` required and vice versa, another could be to have the input fields only appear as the user presses some sort of `Add Witness` button and allow the user to remove them. However, this seemed like the easiest to understand and implement for now

Comment (Yolijn) [reply pages/voorgenomen-huwelijk/getuigen/index.tsx:220]: Ooooh yes good point. Didn't think of the required only when the other is filled in as the purpose here. I agree this is the most readable solution for this usecase then! 😅

**Review by Yolijn — APPROVED** [review]

**Pull request merged**